src/fault_injector/fault_types/platform_blocked_fault.py

```diff
@@ -0,0 +1,27 @@
+from src.component import Component
+from src.fault_injector.fault_types.fault import Fault
+
+
+class PlatformBlockedFault(Fault):
+    """A fault that blocks a platform"""
+
+    @classmethod
+    def from_json(cls, json_object: str) -> "PlatformBlockedFault":
+        """Constructs a PlatformBlockedFault from a JSON object
+
+        :param json_object: The JSON object
+        :type json_object: str
+        :return: a PlatformBlockedFault
+        :rtype: PlatformBlockedFault
+        """
+        raise NotImplementedError()
+
+    def inject_fault(component: Component):
```

Comment:
```suggestion
    def inject_fault(self, component: Component):
```
You probably want a reference to `self`